shared/charts/gpii-flushtokens/templates/cronjob.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  namespace: {{ .Release.Namespace | quote }}
+  name: {{ template "flushtokens.name" . }}
+spec:
+  schedule: "*/15 * * * *"
+  concurrencyPolicy: Allow
+  startingDeadlineSeconds: 200
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            app: {{ template "flushtokens.name" . }}
+        spec:
+          shareProcessNamespace: true
+          containers:
+          - name: gpii-flushtokens
+            image: "{{ .Values.image.repository }}@{{ .Values.image.checksum }}"
+            command: [ "/bin/sh", "-c", "-x", "/app/scripts/dockerDeleteExpiredAccessTokens.sh" ]
+            env:
+            - name: GPII_COUCHDB_URL
+              value: '{{ .Values.couchdbUrl }}'
+            # Use '--deleteAll' to delete all of the access tokens
+            - name: DELETE_ALL
+              value: 'false'
+          - name: istio-proxy-manager
+            image: gcr.io/gke-release/istio/proxyv2:1.0.6-gke.3@sha256:1f6dac3dbf69c0b0d519398003b6572e0218f695f1be295a2ffff95df2395ed0
+            securityContext:
+              allowPrivilegeEscalation: false
+              runAsUser: 1337
+            command: [ "/bin/sh", "-c" ]
+            args:
+              - for i in `seq 1 30`;
+                do
+                  ps -C 'dockerDeleteExpiredAccessTokens.sh' > /dev/null;
```

Comment:
I could never get pgrep to work because it detects this self same script, even when I tried to use a '^' at the beginning of the command name to search for.

A way to get around this issue is to fold these commands into a shell script (e.g, `waitingForGodot.sh`) and then execute that.  But, after reading a fair bit of k8s documentation, I could not see a way to do that.

In the end, I went looking for a command that will find the specific process, based on the process name.  That's what `ps -C` does; in particular, it does not detect the set of commands it's embedded in.

Reply:
@stepanstipl Yes, `ps -C` isn't available using the universal image.  Since `pgrep` finds too many processes, I've looked into other ways of using the version of `ps` available.  I found a way to capitalize on the user id: the rationale is that `node` is the user running the deletion script. For example:
```ps -o comm,user | grep dockerDelete | grep node```

Notes:
- the ps command returns a maximum of 13 characters for the name of the command.  Searching for the full name, "dockerDeleteExpiredAccessTokens.sh" does not work.
- the ps command does not have an option to list full command names in this case.
- normally, you'd use `-u node` with `pgrep` for the same result, but the `-u` option is not available in this case.  That is, `pgrep -u node dockerDelete...` does not work.
- it may be possible to ignore the command name entirely since the only process associated with user `node` is the deletion script.  That is, simply do `ps -o user | grep node`

I have not pushed the script yet, but here is a preview:
```
while ! ps -o comm,user | grep dockerDelete | grep node > /dev/null;
do
  echo 'Waiting for dockerDeleteExpiredAccessTokens.sh to start...';
  sleep 2;
done;
echo 'dockerDeleteExpiredAccessTokens.sh started.';
while  ps -a -o comm,user | grep dockerDelete | grep node > /dev/null;
do
  echo 'Waiting for dockerDeleteExpiredAccessTokens.sh to finish...';
  sleep 2;
done;
echo 'dockerDeleteExpiredAccessTokens.sh finished.';
echo 'Sending TERM to pilot-agent';
pkill pilot-agent;
```

Reply:
I think I like what you committed - without the process name - as it's more generic and can be reused for data loader.  👍

Thanks for explanation btw. :)

Reply:
Your welcome!

I realized that the only task run in the flushtokens container is the deletion script, and the only user running anything is `node`.  Hence, looking for the process's command name is superfluous.

Also, after I make the changes here (`concurrencyPolicy: Replace`, etc.), I'll port the solution to the `dataloader` ([GPII-3999](https://issues.gpii.net/browse/GPII-3999)).